[INFO -- 2018-09-25 03:11:20]
Starting flip_value script
version: 1.1
[INFO -- 2018-09-25 03:11:20]
Initialization strings:
[INFO -- 2018-09-25 03:11:20]
(1) file /tmp/knapsack/ga
[INFO -- 2018-09-25 03:11:20]
initSignal:1
[INFO -- 2018-09-25 03:11:20]
endSignal:3.5
[INFO -- 2018-09-25 03:11:20]
faultModel:3
[EXCEPTION -- 2018-09-25 03:11:22]
pythonException: Unable to exit loop in struct fields; Exiting wihtout making a bit flip
[EXCEPTION -- 2018-09-25 03:11:22]
pythonException: empty range for randrange() (0,0, 0)
[INFO -- 2018-09-25 03:11:22]
Backtrace BEGIN:
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
#1  0x00007ffff7490f5d in __GI_abort () at abort.c:90
#2  0x00007ffff7ae1095 in __gnu_cxx::__verbose_terminate_handler() () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#3  0x00007ffff7adec86 in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#4  0x00007ffff7adecd1 in std::terminate() () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#5  0x00007ffff7adef14 in __cxa_throw () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#6  0x00007ffff7b086f2 in std::__throw_bad_alloc() () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#7  0x000055555555a060 in __gnu_cxx::new_allocator<int>::allocate (this=0x55555577fdf0, __n=18446744072635809791) at /usr/include/c++/7/ext/new_allocator.h:102
#8  0x0000555555559587 in std::allocator_traits<std::allocator<int> >::allocate (__a=..., __n=18446744072635809791) at /usr/include/c++/7/bits/alloc_traits.h:436
#9  0x0000555555558554 in std::_Vector_base<int, std::allocator<int> >::_M_allocate (this=0x55555577fdf0, __n=18446744072635809791) at /usr/include/c++/7/bits/stl_vector.h:172
#10 0x000055555555712e in std::vector<int, std::allocator<int> >::_M_allocate_and_copy<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > > > (this=0x55555577fdf0, __n=18446744072635809791, __first=<error reading variable: Cannot access memory at address 0x100000001>, __last=non-dereferenceable iterator for std::vector) at /usr/include/c++/7/bits/stl_vector.h:1260
#11 0x000055555555617b in std::vector<int, std::allocator<int> >::operator= (this=0x55555577fdf0, __x=std::vector of length -1073741824, capacity -1073741824 = {...}) at /usr/include/c++/7/bits/vector.tcc:206
#12 0x000055555555a1c3 in chromossome::operator= (this=0x55555577fdf0) at knapsackga.cpp:42
#13 0x000055555555a276 in std::__copy_move<false, false, std::random_access_iterator_tag>::__copy_m<chromossome*, chromossome*> (__first=0x555555789b00, __last=0x55555578a000, __result=0x55555577fdf0) at /usr/include/c++/7/bits/stl_algobase.h:324
#14 0x0000555555559854 in std::__copy_move_a<false, chromossome*, chromossome*> (__first=0x555555789b00, __last=0x55555578a000, __result=0x55555577fdf0) at /usr/include/c++/7/bits/stl_algobase.h:386
#15 0x0000555555558c33 in std::__copy_move_a2<false, chromossome*, chromossome*> (__first=0x555555789b00, __last=0x55555578a000, __result=0x55555577fdf0) at /usr/include/c++/7/bits/stl_algobase.h:424
#16 0x0000555555557780 in std::copy<chromossome*, chromossome*> (__first=0x555555789b00, __last=0x55555578a000, __result=0x55555577fdf0) at /usr/include/c++/7/bits/stl_algobase.h:456
#17 0x0000555555556733 in std::vector<chromossome, std::allocator<chromossome> >::operator= (this=0x7fffffffd8d0, __x=std::vector of length 44, capacity 68 = {...}) at /usr/include/c++/7/bits/vector.tcc:223
#18 0x00005555555556c5 in genetic_algorithm () at knapsackga.cpp:117
#19 0x00005555555558d7 in main () at knapsackga.cpp:137

46	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.

Backtrace END
Fliping a bit of the pointer
Memory content before bitflip:0xf00xfd0x770x550x550x550x000x00
Fault Model: Zero bit-flip
base address to flip value: 0x7fffffffd748
address max offset: 8
Memory content after  bitflip:0x000x000x000x000x000x000x000x00
frame name: chromossome::operator=
symbol name: this
symbol filename: knapsackga.cpp
symbol line: 0
value: 0x55555577fdf0
value address: 0x7fffffffd748
Type: The type is a pointer.
Type sizeof: 8

[INFO -- 2018-09-25 03:11:22]
Fault Injection Successful after 2.220500946044922s
[INFO -- 2018-09-25 03:11:22]
event type: exit
[EXCEPTION -- 2018-09-25 03:11:22]
exit code: no exit code available
